core/factorization/lu.cpp

```diff
@@ -40,6 +43,41 @@ GKO_REGISTER_HOST_OPERATION(symbolic_lu_near_symm,
 }  // namespace
 
 
+template <typename ValueType, typename IndexType>
+typename Lu<ValueType, IndexType>::parameters_type
+Lu<ValueType, IndexType>::parse(const config::pnode& config,
+                                const config::registry& context,
+                                const config::type_descriptor& td_for_child)
+{
+    using sparsity_pattern_type = typename experimental::factorization::Lu<
+        ValueType, IndexType>::sparsity_pattern_type;
+    auto params =
+        experimental::factorization::Lu<ValueType, IndexType>::build();
+
+    if (auto& obj = config.get("symbolic_factorization")) {
+        params.with_symbolic_factorization(
+            config::get_stored_obj<const sparsity_pattern_type>(obj, context));
+    }
+    if (auto& obj = config.get("symbolic_algorithm")) {
+        auto str = obj.get_string();
+        if (str == "general") {
+            params.with_symbolic_algorithm(symbolic_type::general);
+        } else if (str == "near_symmetric") {
+            params.with_symbolic_algorithm(symbolic_type::near_symmetric);
+        } else if (str == "symmetric") {
+            params.with_symbolic_algorithm(symbolic_type::symmetric);
+        } else {
+            GKO_INVALID_STATE("Wrong value for symbolic_type");
```

Comment:
nit: the error message for an invalid string value seems to differ every time. Perhaps just fix one blueprint and use that (or maybe a new assertion macro).